dds/idl/v8_generator.cpp

```diff
@@ -156,6 +156,191 @@ namespace {
+        clines += "            continue;\n";
+        clines += "          }\n";
+        clines += "        }\n";
+        clines += "      }\n";
+      } else if (cls & CL_STRING) {
+        clines += "      if (lv->IsString()) {\n";
+        clines += "        v8::Local<v8::String> ls = Nan::To<v8::String>(lv).ToLocalChecked();\n";
+        if (cls & CL_WIDE) {
+          clines += "        std::vector<uint16_t> vwc(ls->Length(), 0);\n";
+          clines += "        ls->Write(&vwc[0]);\n";
+          clines += "        std::wstring ws(ls->Length(), ' ');\n";
+          clines += "        for (size_t i = 0; i < vwc.size(); ++i) {\n";
+          clines += "          ws[i] = vwc[i];\n";
+          clines += "        }\n";
+          clines += "        " + propName + assign_prefix + "ws.c_str()" + assign_suffix + ";\n";
+        } else {
+          clines += "        std::string ss(ls->Length(), ' ');\n";
+          clines += "        ls->WriteUtf8(&ss[0]);\n";
+          clines += "        " + propName + assign_prefix + "ss.c_str()" + assign_suffix + ";\n";
+        }
+        clines += "      }\n";
+      } else if (pt == AST_PredefinedType::PT_char) {
+        clines += "      if (lv->IsString()) {\n";
+        clines += "        v8::Local<v8::String> ls = Nan::To<v8::String>(lv).ToLocalChecked();\n";
+        clines += "        char temp_c;\n";
+        clines += "        ls->WriteUtf8(&temp_c, 1);\n";
+        clines += "        " + propName + assign_prefix + "temp_c" + assign_suffix + ";\n";
+        clines += "      }\n";
+      } else if (pt == AST_PredefinedType::PT_wchar) {
+        clines += "      if (lv->IsString()) {\n";
+        clines += "        v8::Local<v8::String> ls = Nan::To<v8::String>(lv).ToLocalChecked();\n";
+        clines += "        wchar temp_wc;\n";
+        clines += "        ls->Write(&temp_wc, 1);\n";
+        clines += "        " + propName + assign_prefix + "temp_wc" + assign_suffix + ";\n";
+        clines += "      }\n";
+      } else if (pt == AST_PredefinedType::PT_longlong
+              || pt == AST_PredefinedType::PT_ulonglong) {
+        clines += "      if (lv->IsString()) {\n";
+        clines += "        v8::Local<v8::String> ls = Nan::To<v8::String>(lv).ToLocalChecked();\n";
+        clines += "        std::string ss(ls->Length(), ' ');\n";
+        clines += "        ls->WriteUtf8(&ss[0]);\n";
+        clines += "        std::istringstream iss(ss);";
+        clines += "        " + scoped(type->name()) + " temp_ll;\n";
+        clines += "        if (ss.find(\"0x\") != std::string::npos) {\n";
+        clines += "          iss >> std::hex >> temp_ll;\n";
+        clines += "        } else {\n";
+        clines += "          iss >> temp_ll;\n";
+        clines += "        }\n";
+        clines += "        " + propName + assign_prefix + "temp_ll" + assign_suffix + ";\n";
+        clines += "      }\n";
+        clines += "      if (lv->IsNumber()) {\n";
```

Comment:
It might be good to refactor this part (if, Nan::To, assign) since it's very similar for longlong, the other ints, the floats, and boolean.

Reply:
I combined all int-types into a single section, so now all int types have decimal and hex string parsing support. Looking back that this, though, I'm not sure if that's what you meant? I suppose I should add string parsing for float types and bool as well?